frontend/catalyst/api_extensions/control_flow.py

```diff
@@ -539,12 +391,56 @@ class CondCallable:
     released under the Apache License, Version 2.0, with the following copyright notice:
 
     Copyright 2021 The JAX Authors.
+
+    Also provides access to the underlying "Cond" operation object.
+
+    **Example**
+
+
+    .. code-block:: python
+
+        @qml.transform
+        def my_quantum_transform(tape):
+            ops = tape.operations.copy()
+
+            @cond(isinstance(ops[-1], qml.Hadamard))
+            def f():
+                qml.Hadamard(1)
+                return 1
+            @f.otherwise
+            def f():
+                qml.T(0)
+                return 0
+
+            res = f()
+            ops.append(f.operation)
+
+            def post_processing_fn(results):
+                return results
+
+            modified_tape = qml.tape.QuantumTape(ops, tape.measurements)
+            print(res)
+            print(modified_tape.operations)
+            return [modified_tape], post_processing_fn
+
+        @qml.qjit
+        @my_quantum_transform
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def main():
+            qml.Hadamard(0)
+            return qml.probs()
+
+    >>> main()
+    Traced<ShapedArray(int64[], weak_type=True)>with<DynamicJaxprTrace(level=2/1)>
+    [Hadamard(wires=[0]), Cond(tapes=[[Hadamard(wires=[1])], [T(wires=[0])]])]
+    (array([0.25, 0.25, 0.25, 0.25]),)
     """
 
     def __init__(self, pred, true_fn):
         self.preds = [pred]
         self.branch_fns = [true_fn]
         self.otherwise_fn = lambda: None
+        self.operation = None
```

Comment:
I would suggest making this a `@property`, that way we can raise a meaningful error message when the operation doesn't exist, for example because it was called in a classical context :)

Reply:
```suggestion
        self._operation = None
```

A setter is not required since we don't want the property to be overriden outside of this class. Within this class it can be accessed via the "private" attribute `_operation`.

Reply:
Good question about the exception type. An `AttributeError` might make the most sense here?